client.go

```diff
@@ -394,20 +402,56 @@ func (nc *NativeClient) Session(timeout time.Duration) (*ssh.Session, *SessionIn
 	return session, sessionInfo, nil
 }
 
+func (nc *NativeClient) saveSession(client *ssh.Client, sessionInfo *SessionInfo) {
```

Comment:
I agree it needs a new name for our purposes; it was named that way to be consistent with the member openSession type `ssh.Session`. This cleanup I can do as a separate PR right after this one, since this will complicate the review process of the current one. Is that ok?

Reply:
We can leave internal names for later, but let's figure out the names for the public interface functions. Because that's going to percolate out to the local ssh client and the shepherd code.

Reply:
sure, I think `StartPersistentConn` and `StopPersistentConn` are good. The rest is internal functions